frontend_tests/node_tests/compose_ui.js

```diff
@@ -39,7 +39,7 @@ function make_textbox(s) {
     widget.s = s;
     widget.focused = false;
 
-    widget.caret = function (arg) {
+    widget.caret = (arg) => {
         if (typeof arg === "number") {
             widget.pos = arg;
             return this;
```

Comment:
You’ve changed the semantics of this function and several others, because `function` and `=>` bind `this` differently.